### PR TITLE
Solar cooling example

### PR DESCRIPTION
**Pull request by FranziPl**
This is the example of the solar cooling system. Please have a look.

### CODE REVIEW

**Comment by FranziPl** [comment]
Thanks @jakob-wo , I changed things according to your suggestions.

File: solar_cooling/src/electric_model.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+import oemof.solph as solph
+import oemof.solph.processing as processing
+
+import logging
+import os
+import yaml
+import pandas as pd
+import pyomo.environ as po
+
+# import oemof plots
```
Comment (c-moeller) [inline solar_cooling/src/electric_model.py]:
```suggestion
# Import oemof plots
```

File: solar_cooling/src/electric_model.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    else:
+        number_of_time_steps = cfg['number_timesteps']
+
+    solver = cfg['solver']
+    debug = cfg['debug']
+    solver_verbose = cfg['solver_verbose']  # show/hide solver output
+
+    # ## Read data and parameters ## #
+
+    # define the used directories
```
Comment (c-moeller) [inline solar_cooling/src/electric_model.py]:
```suggestion
    # Define the used directories
```

File: solar_cooling/src/electric_model.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+                        param_value['opex_tower']))),
+            bel: solph.Flow()},
+        outputs={bam: solph.Flow()},
+        conversion_factors={
+            bwh: param_value['conv_factor_tower_input_waste'],
+            bel: param_value['conv_factor_tower_input_el']})
+
+    energysystem.add(chil, towe)
+
+    # storages
```
Comment (c-moeller) [inline solar_cooling/src/electric_model.py]:
```suggestion
    # Storages
```

File: solar_cooling/src/electric_results_processing.py [hunk 4]
```diff
@@ -0,0 +1,349 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Dez 06 2018
+
+@author: Franziska Pleissner
+
+System C: concrete example: Plot ocooling process with a solar collector
```
Comment (c-moeller) [inline solar_cooling/src/electric_results_processing.py]:
```suggestion
System C: concrete example: Plot of cooling process with a solar collector
```

File: solar_cooling/src/electric_results_processing.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+import oemof.solph.views as views
+import oemof_visio as oev
+
+import logging
+import os
+import yaml
+import pandas as pd
+from electric_model import ep_costs_func
+
+# import oemof plots
```
Comment (c-moeller) [inline solar_cooling/src/electric_results_processing.py]:
```suggestion
# Import oemof plots
```

File: solar_cooling/src/electric_results_processing.py [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+df_all_var = pd.DataFrame()
+
+
+def electric_postprocessing(config_path, var_number):
+    global df_all_var
+
+    with open(config_path, 'r') as ymlfile:
+        cfg = yaml.load(ymlfile, Loader=yaml.CLoader)
+
+    # define the used directories
```
Comment (c-moeller) [inline solar_cooling/src/electric_results_processing.py]:
```suggestion
    # Define the used directories
```

File: solar_cooling/src/electric_results_processing.py [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+    # Work with the results #
+    #########################
+
+    cool_bus = views.node(energysystem.results['main'], 'cool')
+    waste_bus = views.node(energysystem.results['main'], 'waste')
+    el_bus = views.node(energysystem.results['main'], 'electricity')
+    ambient_res = views.node(energysystem.results['main'], 'ambient')
+    none_res = views.node(energysystem.results['main'], 'None')
+
+    # sequences:
```
Comment (c-moeller) [inline solar_cooling/src/electric_results_processing.py]:
```suggestion
    # Sequences:
```

File: solar_cooling/src/electric_results_processing.py [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+    cool_seq = cool_bus['sequences']
+    waste_seq = waste_bus['sequences']
+    el_seq = el_bus['sequences']
+    ambient_seq = ambient_res['sequences']
+
+    # scalars
```
Comment (c-moeller) [inline solar_cooling/src/electric_results_processing.py]:
```suggestion
    # Scalars
```

File: solar_cooling/src/electric_results_processing.py [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+
+    # Power to the output:
+    electricity_output = el_seq[(('electricity', 'excess_el'), 'flow')].sum()
+    electricity_output_pv = el_seq[(('pv', 'electricity'), 'flow')].sum()
+
+    # ## costs ## #
+
+    costs_total = energysystem.results['meta']['objective']
+
+    # storage costs must be subtract for reference scenario or added
```
Comment (c-moeller) [inline solar_cooling/src/electric_results_processing.py]:
```suggestion
    # Storage costs must be subtract for reference scenario or added
```

File: solar_cooling/src/main.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+
+"""
+Date: 27,05,2020
+Author: Franziska Pleissner
+This App will model a cooling system for Oman. An earlier Version calculated
```
Comment (c-moeller) [inline solar_cooling/src/main.py]:
```suggestion
This oemof application models a cooling system for Oman. An earlier version calculated
```

File: solar_cooling/src/main.py [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+the results of 'Provision of cooling in Oman - a linear optimisation problem
```
Comment (c-moeller) [inline solar_cooling/src/main.py]:
```suggestion
the results of the 'Provision of cooling in Oman - a linear optimisation problem
```

File: solar_cooling/src/main.py [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+with special consideration of different storage options' IRES 2019
```
Comment (c-moeller) [inline solar_cooling/src/main.py]:
```suggestion
with special consideration of different storage options' of the IRES 2019.
```

File: solar_cooling/src/main.py (hunk 10 above)
Comment (c-moeller) [reply solar_cooling/src/main.py]: for a campus building in Oman?

File: solar_cooling/src/main.py (hunk 11 above)
Comment (c-moeller) [reply solar_cooling/src/main.py]: Do you like to add a link to the paper?

File: solar_cooling/src/main.py [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+This version is adapted for oemof 0.4 and uses the solar_thermal_collector from
```
Comment (c-moeller) [inline solar_cooling/src/main.py]:
```suggestion
This version is adapted for oemof v0.4 and uses the solar_thermal_collector from
```

**Review by c-moeller — CHANGES_REQUESTED** [review]
Thank you @FranziPl Looks really good.
I've only made some minimal style comments.

**Comment by FranziPl** [comment]
Thanks @c-moeller . I followed all suggestions.

**Pull request merged**